### ISSUE TITLE
Using variables as type annotations

### ISSUE THREAD

**Issue by daviddavis** [microsoft/pyright#3718]
**Describe the bug**
It seems that pyright and mypy are disagreeing when I use a variable as a type annotation. pyright raises the error `error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)` but the code passes just fine with mypy. 

The real problem is that if I add a `# type:ignore` statement to line in question, then mypy raises `error: Unused "type: ignore" comment` while pyright ignores the line and is happy. So it seems like I cannot satisfy both pyright and mypy.

**Expected behavior**
I would expect consistent results across mypy and pyright. Barring that, is there any guidance on how to make both mypy and pyright happy?

**Screenshots or Code**
```python
from typing import Optional

from pydantic import BaseModel


class UserCreate(BaseModel):
    name: str
    age: int


class UserUpdate(BaseModel):
    __annotations__ = {k: Optional[v] for k, v in UserCreate.__annotations__.items()}
```

**VS Code extension or command-line**
pyright 1.1.261
mypy 0.961

**Comment by tuchandra** [comment]
I don't have any help on the actual type issue here, but to silence pyright alone and not mypy, you can use the comment `# pyright: reportGeneralTypeIssues=false`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.262, which I just published. It will also be included in a future version of pylance.
